### PR TITLE
build(deps): Replaces expo and expo-mail-composer and uses react-native-mail instead

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [PHIRE-426] [PHIRE-467] 

### Description

Removes `expo` and `expo-mail-composer` and replaces it with `react-native-mail`.

Nothing changes in the implementation.

We have the same behavior as before meaning that iOS mail works as before and android as well -

Android cannot attach images and we see the `[OBJ]` square on the mail body **as before**.

But the huge win is that we got rid of expo and its underlying dependencies 

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-426]: https://artsyproduct.atlassian.net/browse/PHIRE-426?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PHIRE-467]: https://artsyproduct.atlassian.net/browse/PHIRE-467?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by damassi — APPROVED** [review]
Worth noting that the current email feature used here is insufficient anyways. One of the big issues is that we don't have the ability to launch email in default browser, that kind of thing. So we should keep that in mind!

**Comment by gkartalis** [comment]
> Worth noting that the current email feature used here is insufficient anyways. One of the big issues is that we don't have the ability to launch email in default browser, that kind of thing. So we should keep that in mind!

The previous implementation wasn't working on android - wasn't able to add the images to the body so it was incomplete eitherway. 

Worth looking into `react-native-mail` as a replacement that might be better.

File: src/screens/Artwork/useMailComposer.ts [hunk 1]
```diff
@@ -1,221 +1,231 @@
...
+    console.log(
+      "[useMailComposer] Sending email:",
+      {
+        subject,
+        ccRecipients,
+      },
+      body
+    )
+  }
+}
```
Comment (damassi) [inline src/screens/Artwork/useMailComposer.ts]: This whole file.... 😅

File: ios/Energy/AppDelegate.h [hunk 2]
```diff
@@ -1,8 +1,7 @@
 #import <React/RCTBridgeDelegate.h>
-#import <Expo/Expo.h>
 #import <UIKit/UIKit.h>
 
-@interface AppDelegate : EXAppDelegateWrapper <UIApplicationDelegate, RCTBridgeDelegate>
```
Comment (brainbicycle) [inline ios/Energy/AppDelegate.h:5]: should this be UIResponder 🤔 will check on a fresh project quickly

Comment (brainbicycle) [reply ios/Energy/AppDelegate.h:5]: yep

**Review by brainbicycle — APPROVED** [review]
🙌 awesome

**Pull request merged**